Solutions/Marain.Tenancy.Cli/Marain/Tenancy/Cli/Commands/List.cs

```diff
@@ -0,0 +1,147 @@
+        /// <param name="tenantProvider">The tenant provider that will be used to retrieve the information.</param>
+        public List(ITenantProvider tenantProvider)
+        {
+            this.tenantProvider = tenantProvider;
+        }
+
+        /// <summary>
+        /// Gets or sets the tenant whose children should be retrieved.
+        /// </summary>
+        [Option(
+            CommandOptionType.SingleOrNoValue,
+            ShortName = "t",
+            LongName = "tenant",
+            Description = "The Id of the tenant to retrieve children for. Leave blank to retrieve children of the root tenant.")]
+        public string? TenantId { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value containing the names of specific properties that should be included in the output.
+        /// </summary>
+        [Option(
+            CommandOptionType.MultipleValue,
+            ShortName = "p",
+            LongName = "property",
+            Description = "The names of tenant properties to include in the output. If omitted, only the tenant Ids will be listed.")]
+#pragma warning disable SA1011 // Closing square brackets should be spaced correctly
+        public string[]? IncludeProperties { get; set; }
+#pragma warning restore SA1011 // Closing square brackets should be spaced correctly
+
+        /// <summary>
+        /// Executes the command.
+        /// </summary>
+        /// <param name="app">The current <c>CommandLineApplication</c>.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task OnExecute(CommandLineApplication app)
+        {
+            if (string.IsNullOrEmpty(this.TenantId))
+            {
+                this.TenantId = this.tenantProvider.Root.Id;
+            }
+
+            string? continuationToken = null;
+
+            var childTenantIds = new List<string>();
+
+            do
+            {
+                TenantCollectionResult children = await this.tenantProvider.GetChildrenAsync(
+                    this.TenantId,
+                    20,
+                    continuationToken).ConfigureAwait(false);
+
+                childTenantIds.AddRange(children.Tenants);
+
+                continuationToken = children.ContinuationToken;
+            }
+            while (!string.IsNullOrEmpty(continuationToken));
+
+            if (this.IncludeProperties?.Length > 0)
+            {
+                await this.LoadAndOutputTenantDetailsAsync(childTenantIds, app.Out).ConfigureAwait(false);
+            }
+            else
+            {
+                this.OutputTenantIds(childTenantIds, app.Out);
+            }
+        }
+
+        private async Task LoadAndOutputTenantDetailsAsync(List<string> children, TextWriter output)
+        {
+            IEnumerable<Task<ITenant>> detailsTasks = children.Select(x => this.tenantProvider.GetTenantAsync(x));
+
+            await Task.WhenAll(detailsTasks).ConfigureAwait(false);
```

Comment:
(was this code in the earlier commit? I don't recognize it.)

Consider assigning the result of this to a local variable. IIRC `WhenAll` should be returning a `Task<ITenant[]>` here, so with that `await` you should be able to assign this into a plain `ITenant[]`. This would mean that a few lines later on, instead of doing `details.Tasks.ForEach` you could run directly against the results, meaning you could avoid the need to write `task.Result` in there.

I know that `task.Result` happens to be OK because you've already awaited all the tasks right here so you know they're all complete, but I still had to do the mental work to discover that when reading the code. Since use of `Result` is frowned upon, and given that it would be pretty easy not to use it here, I'd prefer to avoid it.

Reply:
It was, and I have made the suggested change.